main.js

```diff
@@ -56,6 +57,24 @@ passport.deserializeUser(async (id, cb) => {
   cb(null, user);
 });
 
+app.use(async (ctx, next) => {
+  if (!ctx.session.flash) {
+    ctx.session.flash = {};
+  }
+  ctx.request.flash = (type, msg) => {
+    ctx.session.flash[type] = msg;
+  }
+  ctx.flash = ctx.session.flash;
+  ctx.session.flash = {};
+  ctx.state.flash = ctx.flash;
+
+  await next();
+
+  if (ctx.status === 302 && ctx.session && !(ctx.session.flash)) {
+    ctx.session.flash = ctx.flash;
+  }
+});
```

Comment:
Yay! One thing: how about moving this middleware to its own file? e.g. so you can just have:

```javascript
const flash = require('flash');
app.use(flash);
```

That will help keep `main.js` from getting too big, and make it easier to open source your flash middleware!